lib/magical/parser/duration_parser.ex

```diff
@@ -0,0 +1,21 @@
+defmodule Magical.Parser.DurationParser do
+  @moduledoc """
+  Responsible for parsing durations or delegating to
+  datetime parsing.
+  """
+
+  alias Magical.Parser.DateParser
+
+  def parse("P" <> _rest = duration_string, _args) do
+    Duration.from_iso8601!(duration_string)
```

Comment:
The convention that I created within the `DateParser` was that if a field was not parseable that it should be left as `nil`
```suggestion
    Duration.from_iso8601(duration_string)
    |> case do
      {:ok, duration} -> duration
      error -> nil
    end
```